hnn_core/basket.py

```diff
@@ -22,6 +31,8 @@ def __init__(self, gid, pos, cell_name='Basket'):
         # for height and xz plane for depth. This is opposite for model as a
         # whole, but convention is followed in this function ease use of gui.
         self.shape_soma()
+        self.synapses = dict()
+        self.sect_loc = dict(proximal=['soma'], distal=[])
```

Comment:
`self.sect_loc` shouldn't be defined at the `BasketSingle` level, but rather separately for `L2Basket` and `L5Basket` respectively.

Reply:
I think the reason this is working without throwing any test errors is because by forcing all `feed_loc` variables to be proximal, we're forcing `parreceive_ext` to attempt running `self._connect_feed_at_loc()` if the layer-specific basket cell is defined in `p_src`.

Reply:
Done. Moving forward, we may want to reduce the `if-else` clauses. They make the code really hard to follow and prone to bugs when changing. Thanks for carefully reviewing it.